src/SourceGenerators/Uno.UI.SourceGenerators/BindableTypeProviders/DependencyObjectAvailabilityGenerator.cs

```diff
@@ -141,6 +148,51 @@ internal void Generate(GeneratorExecutionContext context)
 				}
 			}
 
+			private HashSet<INamedTypeSymbol> FindAdditionalLinkerHints(INamedTypeSymbol? additionalLinkerHintAttributeSymbol, GeneratorExecutionContext context)
+			{
+				HashSet<INamedTypeSymbol> types = new();
+
+				if (additionalLinkerHintAttributeSymbol != null)
+				{
+					var attributes = context
+						.Compilation
+						.Assembly
+						.GetAttributes()
+						.Where(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, additionalLinkerHintAttributeSymbol));
+
+					foreach (var attribute in attributes)
+					{
+						if (attribute.ConstructorArguments.FirstOrDefault().Value is string targetType)
+						{
+							if (context.Compilation.GetTypeByMetadataName(targetType) is { } targetSymbol)
+							{
+								types.Add(targetSymbol);
+							}
+							else
+							{
+								var diagnostic = Diagnostic.Create(
+									XamlCodeGenerationDiagnostics.GenericXamlErrorRule,
+									null,
+									$"Failed to find type {targetType} in the current context.");
+
+								context.ReportDiagnostic(diagnostic);
```

Comment:
This works only on netstandard2.0 (Roslyn generators), I think?

Reply:
Correct, but since we're only using Roslyn generators here, and that only net6+ targets can use linker hints, it's fine.

Reply:
@jeromelaban I think the generator is currently included for both netstandard2.0 and net461

Should we wrap the whole file with `#if NETSTANDARD2_0`?